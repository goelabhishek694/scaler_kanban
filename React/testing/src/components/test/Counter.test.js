import { render, screen, fireEvent } from "@testing-library/react";
import Counter from "../Counter";

describe("Counter Component", () => {
  test("initial state check", () => {
    //rendering of comp in isloation
    render(<Counter />);

    //selection
    const counterText = screen.getByText("Count is 0");
    const plusText = screen.getByText("+");
    const minusText = screen.getByText("-");

    // verifying
    expect(counterText).toBeInTheDocument();
    expect(plusText).toBeInTheDocument();
    expect(minusText).toBeInTheDocument();
  });

  test("inc by 1", () => {
    //rendering of comp in isloation
    render(<Counter />);

    //selection    
    const plusText = screen.getByText("+");
    fireEvent.click(plusText);
    
    const isOnePresent = screen.getByText("Count is 1");
    // verifying
    expect(isOnePresent).toBeInTheDocument();
  });

  test("dec by 1", () => {
    //rendering of comp in isloation
    render(<Counter />);

    //selection    
    const minusText = screen.getByText("-");
    fireEvent.click(minusText);
    
    const isMinusOnePresent = screen.getByText("Count is -1");

    // verifying
    expect(isMinusOnePresent).toBeInTheDocument();

  })

});
