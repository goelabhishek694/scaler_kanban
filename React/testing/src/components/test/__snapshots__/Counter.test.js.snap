// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component snapshot for counter 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Counter
    </h1>
    <button>
      +
    </button>
    <p>
      Count is 0
    </p>
    <button>
      -
    </button>
  </div>
</DocumentFragment>
`;
